agent/consul/snapshot_endpoint_test.go

```diff
@@ -120,13 +125,20 @@ func verifySnapshot(t *testing.T, s *Server, dc, token string) {
 	})
 
 	// Restore the snapshot.
-	args.Op = structs.SnapshotRestore
-	restore, err := SnapshotRPC(s.connPool, s.config.Datacenter, s.config.RPCAddr, false,
-		&args, snap, &reply)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
-	defer restore.Close()
+	retry.Run(t, func(r *retry.R) {
+		args = structs.SnapshotRequest{
+			Datacenter: dc,
+			Token:      token,
+			Op:         structs.SnapshotRestore,
+		}
+
+		restore, err := SnapshotRPC(s.connPool, s.config.Datacenter, s.config.RPCAddr, false,
+			&args, snap, &reply)
+		if err != nil {
+			r.Fatalf("err: %v", err)
+		}
+		defer restore.Close()
+	})
```

Comment:
Do we know why this is flaky?

Reply:
Ah, good catch. There's no issues here, as far as I can tell. The autopilot change fixed the flakiness. 

I'll restore this function to the way it was.